BlueprintUI/Sources/Layout/Conditionals.swift

```diff
@@ -0,0 +1,133 @@
+    /// myElement.if(someBoolean) { element in
+    ///     element.centered()
+    /// }
+    /// ```
+    public func `if`(
+        _ isTrue : Bool,
+        then : (Self) -> Element
+    ) -> Element
+    {
+        if isTrue {
+            return then(self)
+        } else {
+            return self
+        }
+    }
+    
+    /// Returns a new element from the provided `then`
+    /// closure if the provided boolean is true. If the provided boolean
+    /// is false, the `else` closure is used
+    /// ```
+    /// myElement.if(
+    ///     someBoolean,
+    ///     then: { element in
+    ///           element.aligned(horizontally: .trailing, vertically: .fill)
+    ///     },
+    ///     else: { element in
+    ///         element.aligned(horizontally: .leading, vertically: .fill)
+    ///     }
+    /// )
+    /// ```
+    public func `if`(
+        _ isTrue : Bool,
+        then : (Self) -> Element,
+        else : (Self) -> Element
+    ) -> Element
+    {
+        if isTrue {
+            return then(self)
+        } else {
+            return `else`(self)
+        }
+    }
+    
+    //
+    // MARK: If Let
+    //
+    
+    /// Returns a new element from the provided `modify`
+    /// closure if the provided value is non-nil. Otherwise, the original
+    /// element is returned.
+    /// ```
+    /// myElement.if(let: someValue) { value, element in
+    ///     element.inset(uniform: someValue.padding)
+    /// }
+    /// ```
+    public func `if`<Value>(
+        `let` value : Value?,
+        then : (Value, Self) -> Element
+    ) -> Element
+    {
+        if let value = value {
+            return then(value, self)
+        } else {
+            return self
+        }
+    }
+    
+    /// Returns a new element from the provided `then`
+    /// closure if the provided boolean is true. If the provided value
+    /// is nil, the `else` closure is used
+    /// ```
+    /// myElement.if(
+    ///     let: someValue,
+    ///     then: { value, element in
+    ///           element.inset(uniform: value.padding)
+    ///     },
+    ///     else: { element in
+    ///         element.inset(uniform: 10)
+    ///     }
+    /// )
+    /// ```
+    public func `if`<Value>(
+        `let` value : Value?,
+        then : (Value, Self) -> Element,
+        else : (Self) -> Element
+    ) -> Element
+    {
+        if let value = value {
+            return then(value, self)
+        } else {
+            return `else`(self)
+        }
+    }
+    
+    //
+    // MARK: Map
+    //
+    
+    /// Creates and returns a new element by passing the
+    /// element to the provided `map` function.
+    ///
+    /// ```
+    /// myElement.map { element in
+    ///     switch myState {
+    ///     case .small: element.inset(uniform: 5)
+    ///     case .medium: element.inset(uniform: 10)
+    ///     case .large: element.inset(uniform: 15)
+    ///     }
+    /// }
+    /// ```
+    public func map(_ map : (Self) -> Element) -> Element
```

Comment:
Didn't we decide _not_ to call this one map, because it's not a container?

Reply:
It's been so long. I honestly forget... I figure this is most familiar to people, so, familiarity probably wins out?

Reply:
I could also see `modify`, but that doesn't quite work, since it's not an in-place transform, you're returning a new element – same with `transform`.